packages/api/src/brevo-config/brevo-config.resolver.ts

```diff
@@ -0,0 +1,103 @@
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { Type } from "@nestjs/common";
+import { Args, ID, Mutation, Query, Resolver } from "@nestjs/graphql";
+
+import { BrevoApiSenderService } from "../brevo-api/brevo-api-sender.service";
+import { BrevoApiSender } from "../brevo-api/dto/brevo-api-sender";
+import { EmailCampaignScopeInterface } from "../types";
+import { DynamicDtoValidationPipe } from "../validation/dynamic-dto-validation.pipe";
+import { BrevoConfigInput, BrevoConfigUpdateInput } from "./dto/brevo-config.input";
+import { BrevoConfigInterface } from "./entities/brevo-config-entity.factory";
+
+export function createBrevoConfigResolver({
+    Scope,
+    BrevoConfig,
+}: {
+    Scope: Type<EmailCampaignScopeInterface>;
+    BrevoConfig: Type<BrevoConfigInterface>;
+}): Type<unknown> {
+    @Resolver(() => BrevoConfig)
+    @RequiredPermission(["brevo-newsletter-config"])
+    class BrevoConfigResolver {
+        constructor(
+            private readonly entityManager: EntityManager,
+            private readonly brevoSenderApiService: BrevoApiSenderService,
+            @InjectRepository(BrevoConfig) private readonly repository: EntityRepository<BrevoConfigInterface>,
+        ) {}
+
+        private async validateSender({ email, name }: { email: string; name: string }): Promise<boolean> {
+            const senders = await this.brevoSenderApiService.getSenders();
+
+            if (!senders || !senders.some((sender) => sender.email === email && sender.name === name)) {
+                return false;
+            }
+
+            return true;
+        }
+
+        @RequiredPermission(["brevo-newsletter-config"], { skipScopeCheck: true })
+        @Query(() => [BrevoApiSender], { nullable: true })
+        async senders(): Promise<Array<BrevoApiSender> | undefined> {
+            const senders = await this.brevoSenderApiService.getSenders();
+            return senders;
+        }
+
+        @Query(() => BrevoConfig, { nullable: true })
+        async brevoConfig(
+            @Args("scope", { type: () => Scope }, new DynamicDtoValidationPipe(Scope))
+            scope: typeof Scope,
+        ): Promise<BrevoConfigInterface | null> {
+            const brevoConfig = await this.repository.findOne({ scope });
+            return brevoConfig;
+        }
+
+        @Mutation(() => BrevoConfig)
+        async createBrevoConfig(
+            @Args("scope", { type: () => Scope }, new DynamicDtoValidationPipe(Scope))
+            scope: typeof Scope,
+            @Args("input", { type: () => BrevoConfigInput }) input: BrevoConfigInput,
+        ): Promise<BrevoConfigInterface> {
+            if (!(await this.validateSender({ email: input.senderMail, name: input.senderName }))) {
+                throw new Error("Sender not found");
+            }
+
+            const brevoConfig = this.repository.create({
+                ...input,
+                scope,
+            });
```

Comment:
Tbh I did not have the sender select in the beginning and just had 2 texts fields (for email & name). That's why I am saving these infos directly. No further reasoning.

But I agree, it would probably make more sense to store the id only in case sender information changes in brevo directly 🤔 

But as you stated correctly this endpoint falls in the "other endpoints" category for the rate limit which is limited to 100 req/HOUR, so I am not sure if storing the values is still the better option in this case
https://developers.brevo.com/docs/api-limits

What do you think?

Reply:
In general, I think it would be better when saving the reference to the brevoId, but because brevo has such a strong rate limiting I think it's better the way you did it.